api/cas-server-core-api-services/src/main/java/org/apereo/cas/services/MultifactorAuthenticationProvider.java

```diff
@@ -24,11 +21,17 @@
     /**
      * Ensure the provider is available.
      *
-     * @param service the service
      * @return true /false flag once verification is successful.
      * @throws AuthenticationException the authentication exception
      */
-    boolean isAvailable(RegisteredService service) throws AuthenticationException;
```

Comment:
I vote that you restore the parameter back. Availability of a provider should be controllable via custom policies linked to a service, allowing for extensions to operate based on service properties, or custom services.

Reply:
The only function I see isAvailable() providing is a basic ping() to the mfa provider to ensure that it is up.  The default code that previously referenced service is now a part of the mfa failure action in the webflow, and perhaps there we should move some of that logic back into the Provider to allow customization without having to provide a custom failure flow action.

I think custom bypass providers are a good place for any decisions that need to be made based on service properties.

My thought is that the details on how to connect to a mfa provider, should not change based on service requested.

Reply:
That does make sense, and your analysis on mfa provider connection capability is correct in theory; but I can see that in practice, we'll surely run into that use case. I would guesstimate the case is not so much about whether mfa should trigger or not (which would make that condition a bypass), but more about how the availability checking of a provider may trigger off custom properties tagged to a service. 

What you have here is fine; but I would vote for the extra flexibility specially since it seems like it won't cost that much, API-wise.